src/functionality/cart.js

```diff
@@ -12,19 +12,28 @@ class Cart {
     return this._cart.findIndex(cartItem => cartItem.productId === productId);
   }
 
+  // Cart is valid if all item contain at least productId and quantity
   _isCartValid() {
+    let valid = true;
     this._cart.forEach((item) => {
```

Comment:
This loop shouldn't be a `forEach`, as forEach is not breakable.

I did not test it, but I think you can use something like:
```javascript
return !this._cart.some((item) => {
    return (!item.productId || !item.quantity);
});
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
I agree